files/default/neutron-evacuate-lbaasv2-agent.py

```diff
@@ -0,0 +1,224 @@
+            cfg.IntOpt('agent_down_time', default=75),
+            cfg.BoolOpt('use_crm',
+                        default=False,
+                        help='When restarting an agent, assume a pacemaker '
+                        'setup and use crm to switch the node into '
+                        'maintenance mode before restarting the agent.'),
+            cfg.BoolOpt('source_agent_restart',
+                        default=True,
+                        help='Restart the neutron-lbaasv2-agent '
+                        'on the node that is being evacuated.'),
+            cfg.BoolOpt('delete_namespaces',
+                        default=False,
+                        help='After evacuating an agent, ssh to it, '
+                        'delete all related loadbalancer namespaces and '
+                        'terminate all processes still running in those '
+                        'namespaces.')
+        ]
+
+        _db_opts = [
+            cfg.StrOpt('connection',
+                       deprecated_name='sql_connection',
+                       default='',
+                       secret=True,
+                       help='URL to database'),
+        ]
+
+        cfg.CONF.register_cli_opts(_db_opts, 'database')
+        cfg.CONF.register_cli_opts(_default_opts)
+        logging.register_options(cfg.CONF)
+        logging.set_defaults()
+
+        self.host_to_evacuate = ""
+
+    # Find all possible destination agents for our migration. Ignore
+    # agent that have set admin_state_down or did not send  their
+    # heartbeat recently
+    def available_destination_agents(self):
+        ret = self.connection.execute(
+            "SELECT id,host,heartbeat_timestamp "
+            "FROM agents WHERE "
+            "agents.host != %s "
+            "AND agents.binary='neutron-lbaasv2-agent' "
+            "AND agents.admin_state_up=true;",
+            self.host_to_evacuate)
+
+        return [{'id': id, 'host': host}
+                for id, host, timestamp in ret.fetchall()
+                if not timeutils.is_older_than(timestamp,
+                                               cfg.CONF.agent_down_time)]
+
+    def loadbalancers_on_agent(self, agent_hostname):
+        ret = self.connection.execute(
+            "SELECT loadbalancer_id "
+            "FROM lbaas_loadbalanceragentbindings,agents "
+            "WHERE agents.host = %s "
+            "AND agents.binary='neutron-lbaasv2-agent' "
+            "AND agents.id=lbaas_loadbalanceragentbindings.agent_id;",
+            agent_hostname)
+        return [id[0] for id in ret.fetchall()]
+
+    def reassign_loadbalancers(self, load_balancer_ids, target_agents):
+        for agent in target_agents:
+            agent['load'] = len(self.loadbalancers_on_agent(agent['host']))
+
+        agents_to_restart = set()
+        for loadbalancer in load_balancer_ids:
+            agent = min(target_agents, key=lambda x: x['load'])
```

Comment:
> An improvement for this would be to split the reassigned load balancers among the target agents. I.e. three agents (A, B, C) with a number of LB (100, 1, 2). In this case if we want to reassign A it will end up in B with 101 LB. Better would be B(51) and C(52).

Actually the code does distribute the loadbalancers equally among the available agents. It updates the load of the agent after each assignment (line 125) and get the new minimum for each loadbalancer